src/components/facets/FacetConnected.tsx

```diff
@@ -9,7 +9,7 @@ import {IFacetState} from './FacetReducers';
 
 const mapStateToProps = (state: IReactVaporState, ownProps: IFacetOwnProps): IFacetStateProps => {
     const item: IFacetState = _.findWhere(state.facets, {facet: ownProps.facet.name});
-    const selectedFacetRows = item ? ownProps.facetRows.filter((row) => _.findWhere(item.selected, {name: row.name})) : [];
+    const selectedFacetRows: IFacet[] = item ? getSelectedRows(ownProps.facetRows, item.selected) : [];

```

Comment:
I don't understand that change... Why didn't the previous line work? It seems like your new way of filtering the selected rows does exactly the same, but with more lines of code. I may be missing something.

Reply:
Is about the format of the selected rows in state. It's not always similar to facetsRow in props especially when they are excluded. So I need to extend the facetRow with the selectedRow if match finds.